### PR TITLE
Truncate Device List In Compatibility Accordion 

### PR DESCRIPTION
**Pull request by danielcliu-ifixit**
## Overview
There are pages where theres a million devices under a compatible product list. This pull truncates that list so its only 3, and makes the page cleaner.

## QA
Does the page `/products/HP-Pavilion-G4-G6-and-G7-Fan` compatibility accordion now truncate into a readable form?
Screenshot from 2022-10-27 17-09-53

Connects: #931

### CODE REVIEW

**Comment by dhmacs** [comment]
@danielcliu-ifixit can we add below the list ".and 23 more" or something like that to hint at the fact that we're truncating the list?

**Comment by jordycosta** [comment]
QA 🟢 

The compatible devices shown in the drop-down are now limited to display 3 devices maximum

sidebar


deploy_block 🟨 on [Massimo's comment](https://github.com/iFixit/react-commerce/pull/935#issuecomment-1294827375)

**Comment by sterlinghirsh** [comment]
The rule should be such that we don't display "show more" if "show more" is only replacing one item. I hate when someone shows me a truncated list with a show more button and it's like why didn't you take the space that was used by the show more button and show me the one remaining item without me having to click.

Something like `if (variants.length > truncate + 1 ) { remaining = variants.length - truncate; variants = variants.slice(0, truncate) } `

**Comment by sterlinghirsh** [comment]
dev_block :zap: un_deploy_block :zap:

**Comment by danielcliu-ifixit** [comment]
un_dev_block :v:

**Comment by deltuh-vee** [comment]
dev_block ✊ 
It looks like we're showing "And 1 other devices..." for a bunch of devices that shouldn't have them in the Compatibility section.
This branch:
image
Master:
image

**Comment by deltuh-vee** [comment]
QA 🎬 
The compatibility section is now truncated to be more readable.
un_dev_block 👍

File: frontend/components/common/CompatibleDevice.tsx [hunk 1]
```diff
@@ -1,11 +1,22 @@
...
 };
 
-export function CompatibleDevice({ device }: CompatibleDeviceProps) {
+export function CompatibleDevice({
+   device,
+   truncate = 0,
+}: CompatibleDeviceProps) {
+   let variants = device.variants;
+   let unlistedVariants = 0;
+   if (truncate > 0 && variants.length > truncate) {
```
Comment (masonmcelvain) [inline frontend/components/common/CompatibleDevice.tsx:16]: Based on Sterling's comment https://github.com/iFixit/react-commerce/pull/935#issuecomment-1295396340, I'd expect this to look like `variants.length > truncate + 1` to handle the case where we only have one extra variant. Maybe I'm missing something and it's being handled elsewhere.

**Review by masonmcelvain — APPROVED** [review]
CR ✌🏻 Nice! deploy_block ✌🏻 on a question about the expected behavior

File: frontend/components/common/CompatibleDevice.tsx (hunk 1 above)
Comment (danielcliu-ifixit) [reply frontend/components/common/CompatibleDevice.tsx:16]: im not quite sure what you mean, my solution is just maxed at `truncate` variants, whiles sterlings is `truncate` + 1 variants.  I felt like `truncate` being the cap made more intuitive sense.

Comment (masonmcelvain) [reply frontend/components/common/CompatibleDevice.tsx:16]: Oh I see, `truncate` doesn't indicate how many variants to show _after_ truncation, it indicates the max number of variants to show before we start truncating.

**Comment by masonmcelvain** [comment]
un_deploy_block ✌🏻

**Pull request merged**